sdk/python/pyproject.toml

```diff
@@ -1,11 +1,11 @@
 [project]
   name = "pulumi_kafka"
   description = "A Pulumi package for creating and managing Kafka."
-  dependencies = ["parver>=0.2.1", "pulumi>=3.0.0,<4.0.0", "semver>=2.8.1"]
+  dependencies = ["parver>=0.2.1", "pulumi", "semver>=2.8.1"]
```

Comment:
Is it intentional that we lost the explicit version for `pulumi`?

Reply:
These changes come from the edits to resources.go I believe

Reply:
This is coming from the default behaviour in the Python language codegen. I too am a little suprised by this. I'll raise it as an issue with core to see if we want to address this before rolling it out further. Either they can address it their side, or we can put the manual version range back in ourselves.

It looks like this is implemented here: https://github.com/pulumi/pulumi/blob/82c546d8731333fa25085ab3a7131971c021941a/pkg/codegen/python/gen.go#L3166-L3170

Which then short-curcuits so it doesn't hit this line which then sets a default version range: https://github.com/pulumi/pulumi/blob/82c546d8731333fa25085ab3a7131971c021941a/pkg/codegen/python/gen.go#L3174-L3175

Reply:
Opened PR for discussion with core: https://github.com/pulumi/pulumi/pull/16151

Reply:
Reverted this for the time being